OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  example.cpp                           *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/Common/DataTable.h>
+
+// Function to do nothing more than  suppress 'un-used variable' warning. 
+// Ignore these calls.
+template<typename AnyType>
+void ignore(AnyType&) {}
+
+
+// This file demonstrates the use of the OpenSim::DataTable_ class. 
+// See OpenSim::DataTable_ for details on the interface. 
+// OpenSim::DataTable_ is a container of a matrix of elements(of any type)
+// that supports random-access to the columns and rows through the their
+// index. Columns can be labeled with std::string and accessing the columns
+// through labels is constant-time on average. DataTable_ can also hold key-
+// value pairs as metadata where key is a std::string and value can be of any
+// type.
+
+int main() {
+  constexpr double EPSILON{0.00001};
+
+  // Default construct a DataTable of SimTK::Real[alias for double].
+  OpenSim::DataTable_<SimTK::Real> dt_real{};
```

Comment:
> convention for myself after C++11 to use {} everywhere for construction/initialization

I think it is harder to read, but also I think to make it a real convention would require things like
```
int i{}, j{}, k{};
```
Probably a better solution is to attempt to strictly follow Stroustrup's RAII convention and avoid declaring variables without initializing them. That is easier in C++11 with move constructors because we can efficiently use function return values to initialize things rather than have to declare variables and then pass them as reference arguments.